controllers/user/password.js

```diff
@@ -0,0 +1,240 @@
+                    let resetUrl = `${req.protocol}://${req.headers.host}${resetPath}?token=${token}`;
+
+                    mail.send(
+                        'Reset the password for your Big Lottery Fund website account',
+                        `Please click the following link to reset your password: ${resetUrl}`,
+                        email
+                    );
+
+                    // mark this user as in password reset mode
+                    models.Users
+                        .update(
+                            {
+                                is_password_reset: true
+                            },
+                            {
+                                where: {
+                                    id: user.id
+                                }
+                            }
+                        )
+                        .then(() => {
+                            req.flash('passwordRequestSent', true);
+                            req.session.save(() => {
+                                res.redirect(userBasePath + userEndpoints.login);
+                            });
+                        })
+                        .catch(err => {
+                            console.error('Error marking user as in password reset mode', err);
+                            req.flash('genericError', 'There was an error requesting your password reset');
+                            return renderUserError(null, req, res, makeUserLink('resetpassword'));
+                        });
+                }
+            })
+            .catch(err => {
+                // error on user lookup
+                console.error('Error looking up user', err);
+                req.flash('genericError', 'There was an error fetching your details');
+                return renderUserError(null, req, res, makeUserLink('resetpassword'));
+            });
+    }
+};
+
+const updatePassword = (req, res) => {
+    let changePasswordToken = req.body.token;
+
+    if (!changePasswordToken) {
+        res.redirect(userBasePath + userEndpoints.login);
+    } else {
+        // @TODO is there a better way to preserve this token between page changes?
+        const currentUrl = makeUserLink('resetpassword') + '?token=' + req.body.token;
+        const errors = validationResult(req);
+
+        if (!errors.isEmpty()) {
+            // failed validation
+            // return the user to the form to correct errors
+            req.flash('formErrors', errors.array());
+            req.session.save(() => {
+                res.redirect(currentUrl);
+            });
+        } else {
+            // check the token again
+            jwt.verify(changePasswordToken, secrets['user.jwt.secret'], (err, decoded) => {
+                if (err) {
+                    console.error('Password reset token expired', err);
+                    req.flash('genericError', 'Your password reset period has expired - please try again');
```

Comment:
This is a general comment that I'm unsure about the use of flash messages for this (and in general). Is it an option to pass errors from `const errors = validationResult(req)` directly into the template? I feel I might be oversimplifying though.